Bloom.md

```diff
@@ -0,0 +1,341 @@
+                in case of non scaling.
+* Bloom Filter: A single bloom filter (Implemented by an external crate).
+* Capacity: The number of items we expect to add to a bloom filter. This controls the size of the filter.
+* False Positive Rate: The accuracy the user expects when operating (set / check) on a filter. This controls the number
+                        of hash functions.
+* Expansion Rate: This is used in scalable bloom filters where multiple bloom filters are stacked to allow users to
+                    continue using the same bloom object when it reaches capacity by adding another filter of larger
+                    capacity (expansion rate * prev filter capacity).
+* Hash Functions: Hash functions used for bit check and set operations underneath.
+* Hash Key: Keys used by the hashing function.
+
+
+### Module OnLoad
+
+Upon loading, the module registers a new bloom module based data type, creates bloom filter (BF.*) commands,
+bloom specific configurations and the bloom ACL category.
+
+* Module name: bloom
+* Data type name: bloom
+
+### Persistence
+
+ValkeyBloom implements persistence related Module data type callbacks for the Bloom data type:
+
+* rdb_save: Serializes bloom objects to RDB.
+* rdb_load: Deserializes bloom objects from RDB.
+* aof_rewrite: Emits commands into the AOF during the AOF rewriting process.
+
+### RDB Compatibility with ReBloom:
+
+During RDB Save of a bloom object, we will need to save the number of filters, expansion rate, false positive rate.
+And for every underlying bloom filter in this object, we store its hash keys used by hashing functions, number of
+hashing functions, number of bits of the bit array, bytes of the bit array itself.
+
+The data above is written into the RDB during save operations and can be restored from RDB load. However, the data that
+gets written to the RDB is specific to the data type's structure and struct members. Additionally, the data within the
+underlying bloom filter (from the external crate) is specific to the implementation of the bloom filter as the
+hash key (seed), hashing algorithm, raw bit array data, etc. can all vary.
+
+Because of this, it is not possible to be RDB compatible with ReBloom. It might be possible if the RDB Save/Load of the
+ReBloom module is reverse engineered AND also the bloom filter implementation is changed to follow ReBloom Module's raw
+Bloom Filter structure syntax. Additionally, if there is any RDB change in ReBloom, this reverse engineering + structure
+updates will be needed once again.
+
+For these reasons, ValkeyBloom is not RDB Compatible with ReBloom.
+
+### AOF Rewrite handling:
+
+Module data types (including bloom) can implement a callback function that will be triggered for Bloom objects to rewrites
+its data as command/s. From the AOF callback, we will handle AOF rewrite by saving a RESTORE command with the key, TTL, and
+serialized value of the corresponding bloom object.
```

Comment:
how do we serialize the bitmap? what about hash functions?

Reply:
> how do we serialize the bitmap? 

It is serialized in a byte array format

> what about hash functions?

Hash functions will not be stored in the RDB. Instead, we are using a global static seed from which hash functions are created.